streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -142,6 +146,25 @@ public StreamTask(final TaskId id,
     }
 
     /**
+     * Finds the source of the given topic partition.
+     *
+     * @param partition the topic partition
+     * @return
+     */
+    private SourceNode findSource(final TopicPartition partition) {
```

Comment:
Yeah I think if the logic is not needed in real cases then we should not add that since it may hide some potential bugs. As long as the tests can be covered then we can get rid of the unnecessary logic.